src/main/java/org/betonquest/betonquest/objectives/BlockObjective.java

```diff
@@ -78,4 +127,35 @@ public void stop() {
         HandlerList.unregisterAll(this);
     }
 
+    private boolean checkLocation(final Location loc, final Profile profile) {
+        try {
+            if (location != null) {
+                if (region != null) {
+                    return isInRange(loc, profile);
+                }
+                return loc.getBlock().getLocation().equals(location.getLocation(profile));
+            }
+        } catch (QuestRuntimeException e) {
+            logger.error(instruction.getPackage(), e.getMessage());
+        }
+        return false;
+    }
+
+    private boolean isInRange(final Location loc, final Profile profile) throws QuestRuntimeException {
+        final Location loc1 = location.getLocation(profile);
+        final Location loc2 = region.getLocation(profile);
+        return inBetween(loc1, loc2, loc);
+    }
+
+    private boolean inBetween(final int range1, final int range2, final int pos) {
+        return Integer.min(range1, range2) <= pos && pos <= Integer.max(range1, range2);
+    }
+
+    private boolean inWorld(final Location range1, final Location range2, final Location pos) {
+        return range1.getWorld().equals(range2.getWorld()) && range2.getWorld().equals(pos.getWorld());
+    }
+
+    private boolean inBetween(final Location range1, final Location range2, final Location pos) {
+        return inBetween(range1.getBlockX(), range2.getBlockX(), pos.getBlockX()) && inBetween(range1.getBlockY(), range2.getBlockY(), pos.getBlockY()) && inBetween(range1.getBlockZ(), range2.getBlockZ(), pos.getBlockZ()) && inWorld(range1, range2, pos);
+    }
```

Comment:
The order of the method should be flipped, so that more general methods are further up.

Additionally, having two methods with different purpose but the same name is confusing, at least one of the `inBetween` should be renamed.